src/test/java/net/sf/jabref/logic/util/io/FileBasedTestCase.java

```diff
@@ -0,0 +1,124 @@
+
+import net.sf.jabref.BibtexTestData;
+import net.sf.jabref.Globals;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.preferences.JabRefPreferences;
+import net.sf.jabref.support.DevEnvironment;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeFalse;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class FileBasedTestCase {
+
+    private BibEntry entry;
+    private Path rootDir;
+
+
+    @Before
+    public void setUp() throws IOException {
+        Globals.prefs = mock(JabRefPreferences.class);
+
+        BibDatabase database = BibtexTestData.getBibtexDatabase();
+        entry = database.getEntries().iterator().next();
+
+        rootDir = Files.createTempDirectory("UtilFindFileTest");
+
+
+
+        Path subDir = Files.createDirectory(rootDir.resolve("Organization Science"));
+        Path pdfSubDir = Files.createDirectory(rootDir.resolve("pdfs"));
+
+        assertTrue(Files.exists(Files.createFile(subDir.resolve("HipKro03 - Hello.pdf"))));
+        assertTrue(Files.exists(Files.createFile(rootDir.resolve("HipKro03 - Hello.pdf"))));
+
+        Path pdfSubSubDir = Files.createDirectory(pdfSubDir.resolve("sub"));
+        assertTrue(Files.exists(Files.createFile(pdfSubSubDir.resolve("HipKro03-sub.pdf"))));
+
+        Files.createDirectory(rootDir.resolve("2002"));
+        Path dir2003 = Files.createDirectory(rootDir.resolve("2003"));
+        assertTrue(Files.exists(Files.createFile(dir2003.resolve("Paper by HipKro03.pdf"))));
+
+        Path dirTest = Files.createDirectory(rootDir.resolve("test"));
+        assertTrue(Files.exists(Files.createFile(dirTest.resolve(".TEST"))));
+        assertTrue(Files.exists(Files.createFile(dirTest.resolve("TEST["))));
+        assertTrue(Files.exists(Files.createFile(dirTest.resolve("TE.ST"))));
+        assertTrue(Files.exists(Files.createFile(dirTest.resolve("foo.dat"))));
+
+        Path graphicsDir = Files.createDirectory(rootDir.resolve("graphicsDir"));
+        Path graphicsSubDir = Files.createDirectories(graphicsDir.resolve("subDir"));
+
+        assertTrue(Files.exists(Files.createFile(graphicsSubDir.resolve("HipKro03test.jpg"))));
+        assertTrue(Files.exists(Files.createFile(graphicsSubDir.resolve("HipKro03test.png"))));
+
+    }
+
+    @Test
+    public void testFindAssociatedFiles() {
+
+        List<BibEntry> entries = Collections.singletonList(entry);
+        List<String> extensions = Arrays.asList("jpg", "pdf");
+        List<File> dirs = Arrays.asList(rootDir.resolve("graphicsDir").toFile(), rootDir.resolve("pdfs").toFile());
+
+        when(Globals.prefs.getBoolean(JabRefPreferences.AUTOLINK_EXACT_KEY_ONLY)).thenReturn(false);
```

Comment:
As far as I can see, there is no need to mock `Globals.prefs`. A simple `preferences = mock(...)` and `findAssociatedFiles(..., preferences)` should work.

Is it much work to refactor `findAssociatedFiles` to accept a boolean `AUTOLINK_EXACT_KEY_ONLY` instead of the preference class?